nano/node/vote_processor.cpp

```diff
@@ -25,33 +25,45 @@ nano::vote_processor::vote_processor (nano::active_transactions & active_a, nano
 	rep_crawler (rep_crawler_a),
 	ledger (ledger_a),
 	network_params (network_params_a),
-	max_votes (flags_a.vote_processor_capacity),
-	started (false),
-	stopped (false),
-	thread ([this] () {
+	max_votes (flags_a.vote_processor_capacity)
+{
+}
+
+nano::vote_processor::~vote_processor ()
+{
+	// Thread must be stopped before destruction
+	debug_assert (!thread.joinable ());
+}
+
+void nano::vote_processor::start ()
+{
+	debug_assert (!thread.joinable ());
+
+	thread = std::thread{ [this] () {
 		nano::thread_role::set (nano::thread_role::name::vote_processing);
-		process_loop ();
-		nano::unique_lock<nano::mutex> lock{ mutex };
-		votes.clear ();
-		condition.notify_all ();
-	})
+		run ();
+	} };
+}
+
+void nano::vote_processor::stop ()
 {
-	nano::unique_lock<nano::mutex> lock{ mutex };
-	condition.wait (lock, [&started = started] { return started; });
+	{
+		nano::lock_guard<nano::mutex> lock{ mutex };
+		stopped = true;
+	}
+	condition.notify_all ();
+	if (thread.joinable ())
```

Comment:
We have the function nano::join_or_pass for this pattern.

Reply:
Yeah, I forgot about that small helper.